python/cudf/cudf/core/column/numerical.py

```diff
@@ -128,6 +129,43 @@ def has_nulls(self, include_nan=False):
             self.nan_count != 0 if include_nan else False
         )
 
+    def __setitem__(self, key: Any, value: Any):
```

Comment:
Just so I understand, why did this PR need to include a `__setitem__` implementation?

Reply:
I created a new `NumericalColumn.__setitem__` implementation because we wanted to error on setting boolean values to a float/int column. Adding these checks and slightly different dtype handling needed for these specific cases compelled me to move the implementation to it's own specialized class.